### DISCUSSION TITLE
Support line highlighting for Docusaurus via `code-line-numbers`

### DISCUSSION THREAD

**Discussion by eitsupi** [quarto-dev/quarto-cli#7199] [Feature Requests]
### Description

It would be great if the following line highlighting that Quarto supports in revealjs format also worked in Docusaurus.

````md
```{.python code-line-numbers="6-8"}
import numpy as np
import matplotlib.pyplot as plt

r = np.arange(0, 2, 0.01)
theta = 2 * np.pi * r
fig, ax = plt.subplots(subplot_kw={'projection': 'polar'})
ax.plot(theta, r)
ax.set_rticks([0.5, 1, 1.5, 2])
ax.grid(True)
plt.show()
```
````

https://quarto.org/docs/presentations/revealjs/index.html#line-highlighting

The syntax on Docusaurus is like this:

````md
```jsx {1,4-6,11}
import React from 'react';

function MyComponent(props) {
  if (props.isBar) {
    return <div>Bar</div>;
  }

  return <div>Foo</div>;
}

export default MyComponent;
```
````

https://docusaurus.io/docs/markdown-features/code-blocks#highlighting-with-metadata-string

The problem here is that `revealjs` is hard-coded in the following part where this conversion is performed, and other formats cannot be added later.
It would be great if third party formats could enable this feature.

https://github.com/quarto-dev/quarto-cli/blob/880565f89b0df66b8ae0f69d30e4959e54464f58/src/resources/filters/quarto-pre/line-numbers.lua#L13-L16

**Comment by cderv** [comment]
Thanks for the idea. I think this is easy enough to implement. I'll have a look.

**Reply by cderv** [reply]
Tracked now in
* https://github.com/quarto-dev/quarto-cli/issues/7201
